strapi/strapi
# [DEITS] Schema validation during the import process

### What does it do?

It compares the source schema and the destination schema with a given strategy to check if the import can be done in the strapi destination.

### Why is it needed?

It's needed to make sure that the strapi destination schemas are compatible with the source ones. 

### How to test it?

1. Create an export file with the command `yarn strapi export --encrypt=false --compress=false`
2. Create a metadata.json file with an empty JSON ({}) inside the exported folder that should be in the root of the exported project
3. Zip it with `tar -czvf backup.tar.gz -C strapi-backup configuration entities links media schemas  metadata.json`
4. Import the project with `yarn strapi import ./backup.tar.gz`
5. Check that there isn´t any error thrown saying that the schemas don't match.
6. If you want to check that if there are differences the error is thrown displaying them, change the file `schemas_00000.json` and repeat from step 3. 
7. You will see that an error is being thrown regarding the incompatibility of the schemas. 

### Related issue(s)/PR(s)
